lib/autokey/qtui/dialogs/show_recent_script_errors.py

```diff
@@ -0,0 +1,152 @@
+from PyQt5.QtCore import pyqtSlot, pyqtSignal
+
+from autokey.model import ScriptErrorRecord
+
+from autokey.qtui import common as ui_common
+
+
+logger = __import__("autokey.logger").logger.get_logger(__name__)
+
+
+class ShowRecentScriptErrorsDialog(*ui_common.inherits_from_ui_file_with_name("show_recent_script_errors_dialog")):
+    """
+    This dialogue is used to show errors that caused user Scripts to abort. The ScriptRunner class holds a list
+    with errors, which can be viewed and cleared using this dialogue window.
+    """
+
+    # When switching errors, emit a boolean indicating if further errors are available. The Next button reacts on
+    # this and enables/disables itself based on the boolean value. The connection is defined in the .ui file.
+    has_previous_error = pyqtSignal(bool, name="has_previous_error")
+    # When switching errors, emit a boolean indicating if previous errors are available. The Previous button reacts on
+    # this and enables/disables itself based on the boolean value. The connection is defined in the .ui file.
+    has_next_error = pyqtSignal(bool, name="has_next_error")
+
+    script_errors_available = pyqtSignal(bool, name="script_errors_available")
+
+    def __init__(self, parent):
+        super(ShowRecentScriptErrorsDialog, self).__init__(parent)
+        self.setupUi(self)
+        # This can’t be done in the UI editor, which can only set specific fonts. Just use the system default monospace
+        # font.
+        self.stack_trace_text_browser.setFontFamily("monospace")
+        self.buttonBox.clicked.connect(self.handle_button_box_buttons)
+
+        self.recent_script_errors = QApplication.instance().\
+            service.scriptRunner.error_records  # type: typing.List[ScriptErrorRecord]
+        self.currently_viewed_error_index = 0
+
+    def hide(self):
+        self.script_errors_available.emit(bool(self.recent_script_errors))
+        super(ShowRecentScriptErrorsDialog, self).hide()
+
+    @pyqtSlot(QAbstractButton)
+    def handle_button_box_buttons(self, clicked_button: QAbstractButton):
+        """
+        Used to connect the button presses with logic for 'Reset error list' and 'Discard current error' buttons in
+        the button box at the bottom of the dialogue window.
+        """
+        button_role = self.buttonBox.buttonRole(clicked_button)
+        # The Close is handled internally and simply hides the dialogue, therefore nothing is implemented here.
+        if button_role == QDialogButtonBox.DestructiveRole:  # Discard current error
+            self.remove_currently_shown_error_from_error_list()
+        elif button_role == QDialogButtonBox.ResetRole:  # Clear the error list
+            self.clear_error_list()
+
+    @pyqtSlot()
+    def update_and_show(self):
+        error_count = len(self.recent_script_errors)
+        if error_count:
+            if self.currently_viewed_error_index >= error_count:
```

Comment:
Closing and re-opening the dialogue keeps the currently viewed error (stored in `self.currently_viewed_error_index`).
This one clamps the stored value into the valid range [0, `error_count`). This is currently dead code (as nothing else modifies the list at the moment), but if there is an API call that clears the list, it will become handy and handle that case.